feature-libs/my-account/organization/components/unit/services/unit-list.service.ts

```diff
@@ -17,60 +25,118 @@ import { OrganizationTableType } from '../../shared/organization.model';
 export class UnitListService extends OrganizationListService<B2BUnit> {
   protected tableType = OrganizationTableType.UNIT;
 
+  protected initialExpanded = 1;
+
+  protected expandState = new Map<string, boolean>();
+  protected levelState = new Map<string, number>();
+  protected update$ = new BehaviorSubject(undefined);
+
+  protected expandedNodeMap = {};
   constructor(
     protected tableService: TableService,
-    protected unitService: OrgUnitService
+    protected unitService: OrgUnitService,
+    protected unitItemService: UnitItemService
   ) {
     super(tableService);
+    this.waitForKey();
   }
 
-  protected load(): Observable<EntitiesModel<B2BUnit>> {
-    return this.unitService
+  protected waitForKey() {
+    this.unitService
       .getTree()
-      .pipe(map((raw) => this.convertUnits(raw)));
+      .pipe(
+        withLatestFrom(this.unitItemService.key$),
+        first(([_tree, id]) => Boolean(id))
```

Comment:
Indeed no memory leak here 👍  
I'd still avoid the explicit subscribe here, and use a switchAll inside the load method, to go from the initial key to loading getTree. But nothing fundamental wrong here.